api/filters.go

```diff
@@ -676,12 +686,12 @@ func (api *API) isValidDatasetDimensions(w http.ResponseWriter, ctx context.Cont
 	return true
 }
 
-// validateCantabularDimensionExists checks that dimension exists in Cantabular by searching for it.
+// getCantabularDimension checks that dimension exists in Cantabular by searching for it.
 // If the dimension doesn't exist, or couldn't be retrieved, an error is returned.
-func (api *API) validateCantabularDimensionExists(ctx context.Context, filterID, dimensionName string) error {
+func (api *API) getCantabularDimension(ctx context.Context, filterID, dimensionName string) (gql.Node, error) {
```

Comment:
Any reason not return a *gql.Node (and then return nil on error)?

Reply:
Don't have a strong pref so have changed 👍

I guess generally if a function is responsible for actually creating a type (e.g. making a request and deserialising the response), then I tend to return the type rather than a pointer. I can see an argument for either, but I'm now so used to Rust's stricter ownership rules 🦀.